### DISCUSSION TITLE
How to enter 'space mode'

### DISCUSSION THREAD

**Discussion by jkmpariab** [helix-editor/helix#550] [Q&A] [answered]
The documentation note in order to enter 'space mode' press 'Space' key, but i found no changing effect implying i am in 'space mode' particularly when i want to copy/paste to/from system clipborad.

what i do: pressing space key then p/P

How to copy/paste to/from system clipboard btw?

**Comment by sudormrfbin** [comment, marked as answer]
There is no indication of pending keys or sub modes in the current released version (v0.3), but they've been added to master (#316 and #515) and will be shipped with the next release. You can build from master if you want to try them out.


> How to copy/paste to/from system clipboard btw?

`space+y` and `space+p` should do exactly that. There should be a message in the command line after copying that says `joined and yanked 1 selection(s) to system clipboard`.

**Reply by jkmpariab** [reply]
Just "cargo install"ed helix-term from master
Actually yank/paste to/from system clipboard functionality not working on the master too
space mode notifications are ok on yanking but not on pasting

Do i need some tools preinstalled like xclip or so?

btw my system specs:
uname -a: Linux <HOST> 5.4.0-80-generic #90-Ubuntu SMP Fri Jul 9 22:49:44 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux

**Reply by jkmpariab** [reply]
By installing xclip everything works fine thanks

**Reply by pickfire** [reply]
@jkmpariab Should helix show an error if xclip is not installed?

**Reply by jkmpariab** [reply]
I think as a vim user just documenting this requirement is sufficient.
But as helix tend to be more modern and user friendlier than vim its a good idea to support this feature

**Reply by CBenoit** [reply]
Yeah, a kind of "helix doctor" (neovim has `:checkhealth`) would be useful for all this
